src/vs/workbench/parts/output/browser/outputActions.ts

```diff
@@ -48,6 +49,33 @@ export class ClearOutputAction extends Action {
 	}
 }
 
+export class ToggleOutputScrollLockAction extends Action {
+
+	public static ID = 'workbench.output.action.toggleOutputScrollLock';
+	public static LABEL = nls.localize('toggleOutputScrollLock', "Toggle Output Scroll Lock");
+
+	private toDispose: IDisposable[] = [];
+
+	constructor(id: string, label: string,
+		@IOutputService private outputService: IOutputService) {
+		super(id, label, 'output-action toggle-output-scroll-lock');
+		this.toDispose.push(this.outputService.onActiveOutputChannel(channel => this._setChecked(this.outputService.getActiveChannel().hasScrollLock)));
+	}
+
+	public run(): TPromise<any> {
+		const activeChannel = this.outputService.getActiveChannel();
```

Comment:
What if there is no active output channel (output hidden) and this command is triggered from the command pallete?
You either need a null check here or even better do not register this action as a global workbench action as I do not see people assigning shortcuts to this action or triggering it via command pallete.